src/pages/questions/ProductInformation.jsx

```diff
@@ -38,10 +40,58 @@ const getImagesUrls = (images, barcode) => {
     .map((key) => `${rootImageUrl}/${key}.jpg`);
 };
 
+// src looks like: "https://static.openfoodfacts.org/images/products/004/900/053/2258/1.jpg"
+const getImageId = (src) => {
+  var sliced = src.split("/").at(-1);
+  sliced.replace(/\..+$/, "");
+  const imgid = sliced.slice(0, -4);
+  return Number(imgid);
+};
+
 const ProductInformation = ({ question }) => {
   const { t } = useTranslation();
   const [productData, setProductData] = React.useState({});
   const [hideImages, setHideImages] = React.useState(getHideImages);
+  const [flagged, setFlagged] = React.useState([]);
+
+  const flagImage = (src, barcode) => {
+    const imgid = getImageId(src);
+    try {
+      const response = axios.put(
+        `https://amathjourney.com/api/off-annotation/flag-image/${barcode}`,
+        {
+          mode: "no-cors",
+          imgid: imgid,
+          url: src,
+        }
+      );
+      const updatedImageArray = [...flagged];
+      updatedImageArray[imgid] = true;
+      setFlagged(updatedImageArray);
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const deleteFlagImage = (src, barcode) => {
+    const imgid = getImageId(src);
+    try {
+      const response = axios.delete(
+        `https://amathjourney.com/api/off-annotation/flag-image/${barcode}`,
+        {
+          mode: "no-cors",
+          data: {
+            imgid: imgid,
+          },
+        }
+      );
+      const updatedImageArray = [...flagged];
+      updatedImageArray[imgid] = false;
```

Comment:
based on  the comment above, this could be `const newFlagged = flagged.filter( flaggedImageId => flaggedImageId !== imageId)` and `setFlagged(updatedImageArray)`